extensions/functions_string.yaml

```diff
@@ -197,6 +197,36 @@ scalar_functions:
           dotall:
             values: [ DOTALL_DISABLED, DOTALL_ENABLED ]
         return: "string"
+  -
+    name: regexp_match_substring
```

Comment:
I believe having the same function name twice will cause an issue in the Java client.  Should this alternative substring function have a different name?

Reply:
@EpsilonPrime 
There is an existing case for [count](https://github.com/substrait-io/substrait/blob/1890e6a7814c5161f38a31aba3e284dde1bc79d4/extensions/functions_aggregate_generic.yaml#L4-L26) with same function name twice. I was using that as reference. 

Shall I change that as well?

Reply:
as discussed yesterday it's okay to keep them with duplicate names in the same file (this is like the files used to be)